File: src/main/java/com/example/the_machine/domain/ThreadEntity.java
package com.example.the_machine.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "thread")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ThreadEntity {

  @Id
  private UUID id;

  private String title;

  @Column(nullable = false)
  private Instant createdAt;

  @Column(nullable = false)
  private Instant lastActivityAt;
}

File: src/main/kotlin/com/example/the_machine/domain/ThreadEntity.kt
package com.example.the_machine.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(name = "thread")
data class ThreadEntity(
  val title: String = "",

  @Column(nullable = false)
  val lastActivityAt: Instant
) : BaseEntity()